cmd/pitr/recoverer/recoverer.go

```diff
@@ -333,6 +335,24 @@ func (r *Recoverer) setBinlogs() error {
 		if sourceID != strings.Split(binlogGTIDSet, ":")[0] {
 			continue
 		}
+
+		if len(r.gtid) > 0 && r.recoverType == Transaction {
+			subResult, err := r.db.SubtractGTIDSet(binlogGTIDSet, r.gtid)
+			if err != nil {
+				return errors.Wrapf(err, "check if '%s' is a subset of '%s", binlogGTIDSet, r.gtid)
+			}
+			if subResult != binlogGTIDSet {
+				set, err := getExtendGTIDSet(binlogGTIDSet, r.gtid)
+				if err != nil {
+					return errors.Wrap(err, "get gtid set for extend")
+				}
+				r.gtidSet = set
+			}
+			if len(r.gtidSet) == 0 {
+				continue
+			}
```

Comment:
why we should *not* search r.startGTID in binlog which has stopGTID?

Reply:
here we just skip files witch didn't have stopGTID